twitter_ads/http.py

```diff
@@ -257,7 +260,11 @@ def perform(self):
                     break
                 bytes_start = bytes_read
                 bytes_read += len(bytes)
-                self.__upload_chunk(location, chunk_size, bytes, bytes_start, bytes_read)
+                response = self.__upload_chunk(location, chunk_size, bytes, bytes_start, bytes_read)
+                response_time = int(response.headers['x-response-time'])
+                chunk_size = min_chunk_size * size(self._DEFAULT_CHUNK_SIZE,
+                                                   self._RESPONSE_TIME_MAX,
+                                                   response_time)
```

Comment:
On the POST request to the /1.1/ton/data/ta_partner endpoint, you'll see something like:

```
location: /1.1/ton/data/ta_partner/2417045708/AeEFBiuxuFY2zgt.txt?resumable=true&resumeId=396213
```

On the *last* PUT request, you'll see something like:

```
location: /ta_partner/2417045708/AeEFBiuxuFY2zgt.txt
```

What we do, using the `location` value from line 253, is `return location.split("?")[0]`, keeping everything before the `?`: `/1.1/ton/data/ta_partner/2417045708/AeEFBiuxuFY2zgt.txt`.